pkg/dns/dns_test.go

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dns
+
+import (
+	"fmt"
+	"math"
+	"net"
+	"reflect"
+	"sort"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/miekg/dns"
+	"istio.io/istio/pkg/test/scopes"
+)
+
+type fakeDNSServer struct {
+	*dns.Server
+	ttl     uint32
+	failure bool
+
+	mu sync.Mutex
+	// map fqdn hostname -> successful query count
+	hosts map[string]int
```

Comment:
The value of this map is the suffix of the IP address instead of the `successful query count` in the comment